grader/src/main/java/edu/pdx/cs410J/grader/scoring/ProjectSubmissionXmlConverter.java

```diff
@@ -9,70 +9,71 @@
 import javax.xml.bind.Marshaller;
 import javax.xml.bind.Unmarshaller;
 import java.io.*;
+import java.text.ParseException;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 
 public class ProjectSubmissionXmlConverter {
-  private final JAXBContext xmlContext;
-  private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private final JAXBContext xmlContext;
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
 
-  public ProjectSubmissionXmlConverter() throws JAXBException {
-    xmlContext = JAXBContext.newInstance(ProjectSubmission.class);
-  }
+    public ProjectSubmissionXmlConverter() throws JAXBException {
+        xmlContext = JAXBContext.newInstance(ProjectSubmission.class);
+    }
 
-  public void convertToXml(ProjectSubmission submission, Writer writer) throws JAXBException {
-    Marshaller marshaller = this.xmlContext.createMarshaller();
-    marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
-    marshaller.marshal(submission, writer);
-  }
+    public void convertToXml(ProjectSubmission submission, Writer writer) throws JAXBException {
+        Marshaller marshaller = this.xmlContext.createMarshaller();
+        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+        marshaller.marshal(submission, writer);
+    }
 
-  public ProjectSubmission convertFromXml(Reader reader) throws JAXBException {
-    Unmarshaller unmarshaller = this.xmlContext.createUnmarshaller();
-    return (ProjectSubmission) unmarshaller.unmarshal(reader);
-  }
+    public ProjectSubmission convertFromXml(Reader reader) throws JAXBException {
+        Unmarshaller unmarshaller = this.xmlContext.createUnmarshaller();
+        return (ProjectSubmission) unmarshaller.unmarshal(reader);
+    }
 
-  public static void main(String... args) throws JAXBException {
-    List<String> outFileNames = new ArrayList<>();
-    outFileNames.addAll(Arrays.asList(args));
+    public static void main(String... args) throws JAXBException {
+        List<String> outFileNames = new ArrayList<>();
+        outFileNames.addAll(Arrays.asList(args));
 
-    if (outFileNames.isEmpty()) {
-      usage("Missing project submission out file name");
-    }
+        if (outFileNames.isEmpty()) {
+            usage("Missing project submission out file name");
+        }
 
-    ProjectSubmissionXmlConverter converter = new ProjectSubmissionXmlConverter();
-    outFileNames.stream().map(File::new).forEach(converter::convertOutFileToXml);
-  }
+        ProjectSubmissionXmlConverter converter = new ProjectSubmissionXmlConverter();
+        outFileNames.stream().map(File::new).forEach(converter::convertOutFileToXml);
+}
 
   private void convertOutFileToXml(File outFile) {
+
+     File xmlFile = null;
```

Comment:
Is the only change that you made to this method is to move this variable declaration to the top?  If so, I'm not sure if that's necessary.  This isn't C where you need to declare all variables at the beginning of a function.  In Java, it's preferred to declare variables close to where they are used.

Reply:
I had to move the variable declaration to the top, because the variable scope wasn't sufficient. That was because of the try, catch block. I think, I have to refactor the code to make the unit test cases pass.